File: CPP/src/sum.cpp
#include "ros/ros.h"
#include <CPP/TwoInts.h>
#include "std_msgs/Int16.h"

ros::Publisher sum;

void callback(const CPP::TwoInts::ConstPtr& msg)
{
  std_msgs::Int16 sumMsg;

  sumMsg.data = msg->a + msg->b;

  sum.publish(sumMsg);
}

int main(int argc, char **argv)
{
  ros::init(argc, argv, "nodeSum");
  ros::NodeHandle n;

  sum = n.advertise<std_msgs::Int16>("sum", 1000);
  ros::Subscriber sub = n.subscribe("two_ints", 1000, callback);

  ros::spin();

  return 0;
}

File: Python/sum_publisher/scripts/sum.py
#!/usr/bin/env python  
import rospy

from std_msgs.msg import Int16
from sum_publisher.msg import TwoInts

def callback(data):
	pub.publish(data.a + data.b)

def listener():
	rospy.init_node('solution', anonymous=True)
	rospy.Subscriber("two_ints", TwoInts, callback)
	rospy.spin()

if __name__ == '__main__':
    try:
		pub = rospy.Publisher('sum', Int16, queue_size=10)
		listener()
    except rospy.ROSInterruptException:
        pass